core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -2083,6 +2075,35 @@ public static boolean equalType(String desc0, MutableRel rel0, String desc1,
     return RelOptUtil.equal(desc0, rel0.rowType, desc1, rel1.rowType, litmus);
   }
 
+  /**
+   * Check if filter under join can be pulled up,
+   * when meeting JoinOnCalc of query unify to Join of target.
+   * Such as {@link JoinOnLeftCalcToJoinUnifyRule} <br/>
+   * {@link JoinOnRightCalcToJoinUnifyRule} <br/>
+   * {@link JoinOnCalcsToJoinUnifyRule} <br/>
+   */
+  private static boolean canPullUpFilterUnderJoin(JoinRelType joinType,
+                                                  @Nullable RexNode leftFilterRexNode,
+                                                  @Nullable RexNode rightFilterRexNode) {
```

Comment:
fix this indentation please.

Reply:
@yanlin-Lynn 
Thanks for reminding.
Please review again.